Nos/Views/Components/NostrNavigationStack.swift

```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+
+/// A NavigationStack that knows how to present views to display Nostr entities like `Events` and `Authors`.
+/// Take care not to nest these.
+struct NostrNavigationStack<Content: View>: View {
```

Comment:
I'm not very convinced about this name. It doesn't look precise to me. The fact that it also adds URL and ReplyToNavigationDestination makes me think that it is more Nos related than Nostr.

Reply:
I agree with your other comment; I like NosNavigationStack for the name here. But I'm open to NostrNavigationStack if @mplorentz has a convincing reason.

Reply:
I was already on the fence between `NosNavigationStack` and `NostrNavigationStack`. Updated!

Reply:
Ah, I just remembered what pushed me towards `NostrNavigationStack` was that I'm creating a `NostrNavigationDestination` enum in my next PR, and `NosNavigationDestination` sounds like it's every navigation destination in the app, which it isn't. But we can address that in the next PR.